store/store.go

```diff
@@ -309,20 +326,25 @@ func (s *Store[H]) Append(ctx context.Context, headers ...H) error {
 	var err error
 	// take current write head to verify headers against
 	var head H
-	headPtr := s.writeHead.Load()
-	if headPtr == nil {
+	if headPtr := s.writeHead.Load(); headPtr == nil {
 		head, err = s.Head(ctx)
 		if err != nil {
 			return err
 		}
+		// store header from the disk.
+		gotHead := head
+		s.writeHead.CompareAndSwap(nil, &gotHead)
 	} else {
 		head = *headPtr
 	}
 
+	slices.SortFunc(headers, func(a, b H) int {
```

Comment:
Now before processing headers are sorted by height in asc order. I don't see any problems that might pop-up after adding this line. Any objections?

The only thing that we should do is to document `.Append` that headers will be processed after sorting them.